bundles/org.openhab.binding.haassohnpelletoven/src/main/java/org/openhab/binding/haassohnpelletoven/internal/HaassohnpelletstoveHandler.java

```diff
@@ -0,0 +1,390 @@
+        stopScheduler();
+    }
+
+    private void stopScheduler() {
+        ScheduledFuture<?> job = refreshJob;
+        if (job != null) {
+            job.cancel(true);
+        }
+        refreshJob = null;
+    }
+
+    /**
+     * Start the job refreshing the oven status
+     */
+    private void startAutomaticRefresh() {
+        ScheduledFuture<?> job = refreshJob;
+        if (job == null || job.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+
+                        updateOvenData(null);
+
+                        for (Channel channel : getThing().getChannels()) {
+                            updateChannel(channel.getUID().getId());
+                        }
+                    } catch (Exception e) {
+                        logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+                    }
+                }
+            };
+
+            @SuppressWarnings("null")
+            int period = config.refreshRate;
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, period, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+
+        if (!automaticRefreshing) {
+            logger.debug("Start automatic refreshing");
+            startAutomaticRefresh();
+            automaticRefreshing = true;
+        }
+
+        verifyLinkedChannel(channelUID.getId());
+        updateChannel(channelUID.getId());
+    }
+
+    @Override
+    public void channelUnlinked(ChannelUID channelUID) {
+
+        if (linkedChannels.containsKey(channelUID.getId())) {
+
+            linkedChannels.remove(channelUID.getId());
+        }
+
+        if (linkedChannels.isEmpty()) {
+            automaticRefreshing = false;
+            stopScheduler();
+            logger.debug("Stop automatic refreshing");
+        }
+    }
+
+    private void updateChannel(String channelId) {
+        if (isLinked(channelId)) {
+
+            State state = null;
+            HaassohnpelletstoveJsonData data = serviceCommunication.getOvenData();
+
+            if (data != null) {
+                switch (channelId) {
+                    case CHANNEL_isTemp:
+                        state = new StringType(data.getisTemp());
+                        update(state, channelId);
+                        break;
+                    case CHANNEL_mode:
+                        state = new StringType(data.getMode());
+                        update(state, channelId);
+                        break;
+                    case CHANNEL_prg:
+                        boolean prg = data.getPrg();
+                        if (prg) {
+                            state = OnOffType.ON;
+                        } else {
+                            state = OnOffType.OFF;
+                        }
+                        update(state, channelId);
```

Comment:
Syntactical sugar. Same for below.
```suggestion
                        update(OnOffType.from(data.getPrg()), channelId);
```